exotica_core/include/exotica_core/kinematic_tree.h

```diff
@@ -78,209 +74,202 @@ inline KinematicRequestFlags operator&(KinematicRequestFlags a, KinematicRequest
     return static_cast<KinematicRequestFlags>(static_cast<int>(a) & static_cast<int>(b));
 }
 
-class KinematicFrameRequest
+struct KinematicFrameRequest
 {
-public:
     KinematicFrameRequest();
-    KinematicFrameRequest(std::string frameALinkName, KDL::Frame frameAOffset = KDL::Frame(), std::string frameBLinkName = "", KDL::Frame frameBOffset = KDL::Frame());
-    std::string FrameALinkName;
-    KDL::Frame FrameAOffset;
-    std::string FrameBLinkName;
-    KDL::Frame FrameBOffset;
+    KinematicFrameRequest(std::string _frame_a_link_name, KDL::Frame _frame_a_offset = KDL::Frame(), std::string _frame_b_link_name = "", KDL::Frame _frame_b_offset = KDL::Frame());
+    std::string frame_a_link_name;
+    KDL::Frame frame_a_offset;
+    std::string frame_b_link_name;
+    KDL::Frame frame_b_offset;
 };
 
-class KinematicsRequest
+struct KinematicsRequest
 {
-public:
     KinematicsRequest();
-    KinematicRequestFlags Flags = KinematicRequestFlags::KIN_FK;
-    std::vector<KinematicFrameRequest> Frames;  //!< The segments to which the end-effectors are attached
+    KinematicRequestFlags flags = KinematicRequestFlags::KIN_FK;
+    std::vector<KinematicFrameRequest> frames;  // The segments to which the end-effectors are attached
 };
 
 struct KinematicFrame
 {
-    std::weak_ptr<KinematicElement> FrameA;
-    KDL::Frame FrameAOffset;
-    std::weak_ptr<KinematicElement> FrameB;
-    KDL::Frame FrameBOffset;
-    KDL::Frame TempAB;
-    KDL::Frame TempA;
-    KDL::Frame TempB;
+    std::weak_ptr<KinematicElement> frame_a;
+    KDL::Frame frame_a_offset;
+    std::weak_ptr<KinematicElement> frame_b;
+    KDL::Frame frame_b_offset;
+    KDL::Frame temp_ab;
+    KDL::Frame temp_a;
+    KDL::Frame temp_b;
 };
 
-/**
- * @brief      The KinematicResponse is the container to keep kinematic update
- * data. The corresponding KinematicSolution is created from and indexes into a
- * KinematicResponse.
- */
-class KinematicResponse
+/// @brief The KinematicResponse is the container to keep kinematic update data.
+/// The corresponding KinematicSolution is created from and indexes into a KinematicResponse.
+struct KinematicResponse
 {
-public:
     KinematicResponse();
-    KinematicResponse(KinematicRequestFlags Flags, int Size, int N = 0);
-    KinematicRequestFlags Flags = KinematicRequestFlags::KIN_FK;
-    std::vector<KinematicFrame> Frame;
-    Eigen::VectorXd X;
-    ArrayFrame Phi;
-    ArrayTwist PhiDot;
-    ArrayJacobian J;
-    ArrayJacobian JDot;
+    KinematicResponse(KinematicRequestFlags _flags, int _size, int _N = 0);
+    KinematicRequestFlags flags = KinematicRequestFlags::KIN_FK;
+    std::vector<KinematicFrame> frame;
+    Eigen::VectorXd x;
```

Comment:
X should remain upper case. It is the vector (thus upper case) of configuration values (as in papers)

Reply:
Unresolved